composer/core/state.py

```diff
@@ -144,17 +142,21 @@ class State(Serializable):
     algorithms: Sequence[Algorithm] = tuple()
     callbacks: Sequence[Callback] = tuple()
 
-    # machine info
-    world_size: int = 1
-    nproc_per_node: int = 1
+    @property
+    def world_size(self) -> int:
+        return get_world_size()
 
     @property
     def global_rank(self) -> int:
         return get_global_rank()
 
+    @property
+    def local_world_size(self) -> int:
+        return get_local_world_size()
```

Comment:
Yeah... I've seen a bit of variation wrt which one is used. I marginally prefer `local_world_size` because there's a slim chance we could end up with nodes with different sizes.